src/azure-cli/azure/cli/command_modules/containerapp/_utils.py

```diff
@@ -0,0 +1,2079 @@
+    except:  # pylint: disable=bare-except
+        pass
+    return job
+
+
+def _get_name(name_or_rid):
+    if is_valid_resource_id(name_or_rid):
+        return parse_resource_id(name_or_rid)["name"]
+    return name_or_rid
+
+
+def _get_default_containerapps_location(cmd, location=None):
+    if location:
+        return location
+    default_location = "eastus"
+    providers_client = None
+    try:
+        providers_client = providers_client_factory(cmd.cli_ctx, get_subscription_id(cmd.cli_ctx))
+        resource_types = getattr(providers_client.get(CONTAINER_APPS_RP), 'resource_types', [])
+        res_locations = []
+        for res in resource_types:
+            if res and getattr(res, 'resource_type', "") == "workspaces":
+                res_locations = getattr(res, 'locations', [])
+
+        if len(res_locations) > 0:
+            location = res_locations[0].lower().replace(" ", "").replace("(", "").replace(")", "")
+            if location:
+                return location
+
+    except Exception:  # pylint: disable=broad-except
+        return default_location
+    return default_location
+
+
+# Generate random 4 character string
+def _new_tiny_guid():
+    import random
+    import string
+    return ''.join(random.choices(string.ascii_letters + string.digits, k=4))
+
+
+#  Generate a random volume name using same method as log analytics workspace
+def _generate_secret_volume_name():
+    import re
+    prefix = "secret-volume"
+    # volume name must be lowercase
+    suffix = _new_tiny_guid().lower()
+    maxLength = 40
+
+    name = "{}-{}".format(
+        prefix,
+        suffix
+    )
+
+    if len(name) > maxLength:
+        name = name[:maxLength]
+    return name
+
+
+# Follow same naming convention as Portal
+def _generate_log_analytics_workspace_name(resource_group_name):
+    import re
+    prefix = "workspace"
+    suffix = _new_tiny_guid()
+    alphaNumericRG = resource_group_name
+    alphaNumericRG = re.sub(r'[^0-9a-z]', '', resource_group_name)
+    maxLength = 40
+
+    name = "{}-{}{}".format(
+        prefix,
+        alphaNumericRG,
+        suffix
+    )
+
+    if len(name) > maxLength:
+        name = name[:maxLength]
+    return name
+
+
+def _get_log_analytics_workspace_name(cmd, logs_customer_id, resource_group_name):
+    log_analytics_client = log_analytics_client_factory(cmd.cli_ctx)
+    logs_list = log_analytics_client.list_by_resource_group(resource_group_name)
+    for log in logs_list:
+        if log.customer_id.lower() == logs_customer_id.lower():
+            return log.name
+    raise ResourceNotFoundError("Cannot find Log Analytics workspace with customer ID {}".format(logs_customer_id))
+
+
+def _generate_log_analytics_if_not_provided(cmd, logs_customer_id, logs_key, location, resource_group_name):  # pylint: disable=too-many-statements
+    if logs_customer_id is None and logs_key is None:
```

Comment:
should we remove log analysis generation logic as RP does not require it now? @howang-ms